.github/workflows/integration_test_run.yaml

```diff
@@ -185,7 +193,13 @@ jobs:
         modules: ${{ fromJSON(inputs.modules) }}
         ${{ insert }}: ${{ fromJSON(inputs.extra-test-matrix) }}
       fail-fast: false
-    runs-on: ${{ inputs.runs-on }}
+    runs-on: >-
+      ${{
+        inputs.runs-on == 'ubuntu-22.04' &&
+        inputs.self-hosted-runner &&
```

Comment:
shouldn't it be not inputs.self-hosted-runner?

Reply:
The logic is:
```
if inputs.runs-on == 'ubuntu-22.04' and inputs.self-hosted-runner:
  fromJson(format('[''self-hosted'', ''{0}'']',  inputs.self-hosted-runner-label))
else:
  inputs.runs-on
```
The reason for the `inputs.runs-on == "ubuntu-22.04"` is due to the github-runner charm only supports jammy right now.

Reply:
ah I see, yeh